### ISSUE TITLE
Comments on scoping doc

### ISSUE THREAD

**Issue by AArnott** [ShieldedLabs/crosslink-deployment#6]
From the [scoping doc](https://github.com/ShieldedLabs/crosslink-deployment/blob/main/Scoping.md), a couple questions:

> It reduces supply of ZEC by locking up staked ZEC

The doc doesn't talk at all about _un_locking ZEC. What is that process? Are there set timeframes the user must consent to in advance, or can the ZEC be unlocked at any time?

> reduces required confirmations to a nice small number like 10.

Maybe out of scope for a scoping doc, but I'm curious what challenges exist to shrink this number even further. Re-orgs in excess of 10 blocks should be rare, so maybe that makes finality easier to achieve. But I wonder if we could get to finality within 2-3 blocks eventually.

> Unsophisticated users (who understand little about crypto and do not use specialized tools) delegate ZEC and get rewards from their mobile wallet.

That sounds awesome. I'll be interested in learning what mobile wallet devs need to do, and what UX we'll need to add, to support this.

**Comment by zookoatshieldedlabs** [comment]
> The doc doesn't talk at all about _un_locking ZEC. What is that process? Are there set timeframes the user must consent to in advance, or can the ZEC be unlocked at any time?

Yes, it's a good question, but I don't think it needs to be part of the scoping doc. I suppose we'll learn more about the options and trade-offs of that as we experiment with our prototypes. However, the stated motivation of locking up ZEC is still valid, even if users can unlock ZEC at any time, because staking generates demand for ZEC.

**Comment by arya2** [comment]
> Why Hybrid Proof-of-Work/Proof-of-Stake?
> ---
> * It reduces supply of ZEC by locking up staked ZEC and increases demand for ZEC by giving people something else to do with it. Both reducing supply and increasing demand put upward pressure on the price of ZEC. This is good because the price of ZEC is the fuel for the mission and attracts users.

This point makes sense for PoS, but doesn't explain the motivation for a hybrid PoW/PoS, I would actually interpret it as a motivation for pure PoS. A "Why Proof-of-Stake" subheading might be effective here, as well as some mention of the motivation for hybrid PoW/PoS over pure PoS.

> It reduces supply of ZEC by locking up staked ZEC

I'm not sure that it necessarily does, those coins would be less liquid which could help lower the velocity of the supply (depending on how long staked ZEC would be locked before it can be unstaked), but I'd be skeptical of that having a substantial impact on the supply/demand balance relative to the impact of reducing energy costs required to maintain the chain (that are currently effectively paid for by all ZEC holders), and offering a way of at least partially avoiding dilution/inflation when ZEC is issued via block subsidies.

I think if we want to keep this point, we should make it conditional on the duration that staked ZEC would have to remain locked, and should mention it after noting that it partially compensates ZEC holders for inflation (if we agree that that'll probably have a bigger impact).

> something else to do with it.

I would be more specific here and say that it gives ZEC holders a way to partially avoid inflation as new ZEC is issued via block subsidies.

> * It adds finality, which protects users from being robbed (with hybrid protection), reduces deposit times at CEXes and other services, and enables bridges.

I would say it _enables_ finality, or I would change the header to be more specifically about Crosslink or Trailing Finality.

> * It lays the foundation for future improvements, such as cross-chain interop and scalability.

If I understand correctly, it's the finality that enables cross-chain interop, so I would mention that in the point above (though I think it's already covered by "enables bridges")

> UX goals
> ---
> ....
> * Other services that require finality, such as cross-chain bridges, are willing to rely on Zcash’s finality.

This point doesn't seem like it would directly affect the UX on its own, I think it's enough to mention that finality would enable bridges in the motivation section, though it seems very useful so I would elaborate on why that's useful in the motivation section.

> * Unsophisticated users (who understand little about crypto and do not use specialized tools) delegate ZEC and get rewards from their mobile wallet.

I think it would be better to say that it makes contributing to mining/appending blocks onto the chain and receiving block rewards easier and more accessible for everyone (since it would make it easier for even the most technically proficient users, and technical proficiency isn't currently the only barrier there).

> Requirements
> ---
> * Zcash transactions come with a kind of finality which protects the users as much as possible against all possible attacks, and is sufficient for services such as cross-chain bridges and centralized exchanges.

> Zcash transactions come with a kind of finality

I think we could call this strong or deterministic finality? 

> which protects the users as much as possible against all possible attacks

This seems too broad, I would think it makes _some_ kinds of attacks a little easier, could we say that it offers greater protection rather than the maximum possible degree of protection?

> Trade-offs
> ---
> 
> Goals which we currently believe are not safely achievable in this first deployment without losing some of the above goals and requirements:
> * Improving Zcash's bandwidth (number of transactions per time) or latency (time for a transaction).
> * Deploying cross-chain interoperation such as the Inter-Blockchain Communication protocol (IBC) or Cross-Chain Interoperability Protocol (CCIP).

Specifying what the trade-offs are here could be useful, I expect that pursuing these goals would be at the expense of deploying _some_ version of Crosslink as soon as possible.

> * Improving Zcash's bandwidth

"Throughput" might be the more conventional term here.

**Comment by zookoatshieldedlabs** [comment]
> > reduces required confirmations to a nice small number like 10.
> 
> Maybe out of scope for a scoping doc, but I'm curious what challenges exist to shrink this number even further. Re-orgs in excess of 10 blocks should be rare, so maybe that makes finality easier to achieve. But I wonder if we could get to finality within 2-3 blocks eventually.

Yes!! I think we can! In a subsequent iteration after the initial deployment of Crosslink. Did you see this comment in the "Trade-Offs" section at the end?

> Goals which we currently believe are not safely achievable in this first deployment without losing some of the above goals and requirements:
> 
> Improving Zcash's bandwidth (number of transactions per time) or latency (time for a transaction).

But more details about how much we *could* reduce confirmations and what the attendant trade-offs would be, that would be cool. Maybe we should create a ... A GitHub issue? To track that?

**Comment by zookoatshieldedlabs** [comment]
> > Unsophisticated users (who understand little about crypto and do not use specialized tools) delegate ZEC and get rewards from their mobile wallet.
> 
> That sounds awesome. I'll be interested in learning what mobile wallet devs need to do, and what UX we'll need to add, to support this.

Great! [eZcash](https://blog.nerdbank.net/ezcash-app) would be a great app to use to work out the UX issues.

So, let's sketch out a first draft of a user story:

---

Sam is a Zcash user, who has the `eZcash` wallet and has some ZEC in it. Sam may or may not have heard (from a friend or from reading something on the internet) about "staking" or "proof-of-stake" or "earning yield" or whatever, but if Sam has heard of it, they don't know anything about it and they're not sure they should trust it. It could be, as far as Sam knows, an attempt by some scammer to trick Sam into giving over his ZEC!

So then, Sam notices a new feature in `eZcash`. It could be a message that pops up when they upgrade eZcash, or it could be that they just notice a new button in the eZcash UI.

Here's the critical part: Sam trusts eZcash! The only way he would try this "staking" thing at all is because he figures if eZcash is offering it to him then it must be legit.

Now there are a few things that eZcash needs to help Sam decide, here:

1. How risky is it? If Sam chooses to stake some of his ZEC, could that ZEC be wholly or partially lost?? (Answer: yes. Staking comes with a risk of your ZEC being burned, a.k.a. "slashed", due to error or misbehavior on the part of the validator you used to stake. Historically, this seems to be very rare in Cosmos and Ethereum, but there is no way to guarantee to Sam that his staked ZEC is 100% safe. One guarantee we *can* make is that his ZEC can't be stolen. The validator he stakes with could accidentally or malicious burn his ZEC, but it can't steal his ZEC.)
2. What's the reward? Is it guaranteed in advance, like "If you stake your ZEC, you'll get 5% APR reward for the next 21 days." or is it determined dynamically, like "If you stake your ZEC, you'll probably get somewhere between 0.001% and 100% APR reward 21 days from now."?
3. Based on the above, how much of Sam's ZEC is he going to stake? Is there a slider/dial/number-entry-box for him to choose that? Is there a display showing how much of his ZEC is staked versus how much is in his hot wallet? Or does eZcash integrate staking with eZcash's "accounts" abstraction, like "All ZEC that you move into _this_ account over here is automatically staked."
4. Also, how does the lock-up work? Presumably Sam needs to understand that part 1 and part 2 above are both based on a certain passage of time. Is there just one fixed lock-up period, like "Click here to lock-up X of your ZEC for 21 days in return for the above-mentioned risks and rewards"? Or does Sam get to dial in a time period along with dialing in an amount of ZEC to stake? Or is a certain bucket (account) of Sam's ZEC automatically staked (including getting staked again once the unbonding period elapses)?
5. Which validator is Sam's ZEC going to be staked with? Is there just one validator baked into eZcash (maybe a validator operated by the maintainer of eZcash?), so that Sam isn't even aware of there being an option? Or is there a list of the names and URLs of validators (whitelisted/approved by the maintainer of eZcash) that Sam has to pick from? Or does same need to find a validator's URL, or "address/key", or code, or QR code from somewhere and type or paste that into eZcash to choose a validator? (Remember: _Sam trusts eZcash!_ And he wisely does *not* trust anything he reads on the internet. So his decision procedure is likely going to be: "Use whatever eZcash recommends.". Asking him to do research on the Internet and form trust decisions about other people, orgs, sites, apps, etc. other than eZcash is a huge, huge lift, making him feel (with good reason) that he is putting his money at risk. So there's a huge user experience gap between the different possible answers to this question.)_

Phewf! Okay that's my first rough draft. @AArnott : WHAT YOU DO THINK ?? :-)

**Comment by AArnott** [comment]
You've thought through it quite well, and I like what I see.

Where do the rewards for staking go? Directly back into the pile of staked ZEC, or into Sam's spendable account?
I hope the time period is flexible. And especially that it isn't locked to a 21 day period, since that sounds like it'll be high maintenance if you want to stake long-term.

I buy your value prop for eZcash recommending a trustworthy validator. Although that may set me up for liability, which I'll want to look into (possibly with a lawyer). If I set up my own Validator, that would confine the risks to that Validator suffering an outage, leading to slashed funds, correct?

**Comment by zookoatshieldedlabs** [comment]
> This point makes sense for PoS, but doesn't explain the motivation for a hybrid PoW/PoS, I would actually interpret it as a motivation for pure PoS. A "Why Proof-of-Stake" subheading might be effective here, as well as some mention of the motivation for hybrid PoW/PoS over pure PoS.

This suggestion is integrated into the current version of the scoping doc.

**Comment by arya2** [comment]
> what do you think of the current text about this in the scoping document?

This looks a lot better, I think the second point is perfect, I would change the first point slightly, to:

"Enabling ZEC holders to stake their funds to claim block rewards should encourage demand for ZEC from stakers, and, partially replacing miners that often must sell their mined ZEC to recoup energy costs should generally reduce the supply of ZEC available for purchase."
